src/app/server/Server.cpp

```diff
@@ -270,7 +270,7 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
             .fabricTable       = &mFabrics,
             .clientPool        = &mCASEClientPool,
             .groupDataProvider = mGroupsProvider,
-            .mrpLocalConfig    = Optional<ReliableMessageProtocolConfig>(GetLocalMRPConfig()),
+            .mrpLocalConfig    = GetLocalMRPConfig(),
```

Comment:
As discussed in slack

> I want to give LocalMrpConfig a new type instead of using Optional<MrpConfig> for 2 purpose:
>  1. Enforce its constructor, to prevent regression like this problem
>  2. Save heap memory for platforms which want to use a global unique config. Then the size of this object can by 0, we don't bother passing the config value around.

Anyway, it should be a following up PR, because this PR is already a bit too big.

Reply:
Yes, followup for this part is totally fine; we should just make sure it happens.

Reply:
#19847